pages/index.html

```diff
@@ -10,8 +10,40 @@
   <script src="js/templates/templates.js"></script>
   <script>
     document.addEventListener('DOMContentLoaded', function () {
-      Coveo.SearchEndpoint.configureSampleEndpointV2()
-      Coveo.init(document.body);
+      Coveo.SearchEndpoint.configureSampleEndpointV2();
+      // TODO: remove this great example
+      Coveo.init(document.body, {
+        hooks_demo: {
+          customHeaderRenderer: {
+            renderTitle: facet => {
+              const wrapperElement = document.createElement('span');
+              wrapperElement.className = 'coveo-dynamic-facet-header-title';
+
+              const linkElement = document.createElement('a');
+              linkElement.href = 'https://docs.coveo.com/en/2036/cloud-v2-administrators/understanding-fields#field-types';
+              linkElement.style.color = 'orange';
+              linkElement.style.textDecoration = 'none';
+              linkElement.target = 'blank';
+  
+              const titleElement = document.createElement('h3');
+              titleElement.style.margin = '0';
+              titleElement.innerHTML = `${facet.options.title} 🔗`;
+
+              linkElement.appendChild(titleElement);
+              wrapperElement.appendChild(linkElement);
+              return wrapperElement;
+            },
+
+            renderClearButton: () => {
```

Comment:
Yes yesterday I thought about it being too demanding and started to add the option to only modify the basic element by passing it as an argument, this way a client could modify some parts of it or add hooks and stuff.

We could assert a few things on the method's return that would help a developper, but I wouldn't be too strict as we could let a user do really funky things with the component (for example what if they don't want the button "inline" but maybe outside the facet). The most important thing is to prevent JS errors with our default behaviour, for the rest I would see the implementer of a custom element "responsible" for the consequences of it's own code as they would be for writing a custom one from scratch, but this time with a lot less effort. I'd also except someone going in those options to have some basic HTML/CSS/JS knowledge like the difference between displays.

For backward compatibility, I would say since we don't want to break our own styles, this would also prevent a client's style to his custom elements.

I made sure that the aria-label is correctly added after the custom render methods so they won't have to think about those.

Reply:
I understand the intention to "give freedom to do funky things" and let customers do "what they want" (because that's something we've done a lot on the past), but with hindsight, and if I had to start the project from scratch, I'd say that giving too much liberty means a ** LOT ** more support case/questions, and I would dial back that down a lot, and add much more constraints in some places.

It has been useful sometimes in the past when we said to ourselves "let's make it completely customizable, and implementers will do what they want".

But as also given us a lot of headache, and has been a relatively big burden in other places in the company. 

For examples, those kind of decision have a big impact on the support team, and on the kind of people they will need to recruit to support our clients.

Expecting people to have some basic HTML/CSS/JS is okay. But the variance in that area is... kind of insane. And your definition of basic is definitely not equal to the definition of basic for other people :)

I guarantee you that as soon as we document this, and anything goes wrong with the custom code, even if the Search UI has no bug per say in that part, we are going to hear from it through maintenance case.

For backward compatibility, what you are describing means that, for ever and ever, we won't be able to touch the styling of DynamicFacet. Even for a bugfix (say, in some setup in SFDC, the Facet header does not display properly, and we need to tweak the CSS).

That seems excessively constraining.